src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,198 @@
+import java.util.LinkedList;
+
+import static frc.robot.Ports.Conveyor.INVERSION;
+
+public class Conveyor extends SubsystemBase {
+    private static Conveyor INSTANCE = null;
+    private final WPI_TalonFX motor = new WPI_TalonFX(Ports.Conveyor.MOTOR);
+    private final Deque<String> position = new LinkedList<>();
+    private final ColorSensorV3 colorSensorIntake = new ColorSensorV3(I2C.Port.kOnboard);
+    private final DigitalInput postFlapBeam = new DigitalInput(Ports.Conveyor.POST_FLAP_BEAM_BREAKER);
+    private final DigitalInput preFlapBeam = new DigitalInput(Ports.Conveyor.PRE_FLAP_BEAM_BREAKER);
+    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Conveyor.SOLENOID);
+    private final ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean lastPostFlapBeamInput = true;
+
+    private Conveyor() {
+        motor.setInverted(INVERSION);
+        match.addColorMatch(Constants.Conveyor.RED);
+        match.addColorMatch(Constants.Conveyor.BLUE);
+        match.addColorMatch(Constants.Conveyor.NONE);
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * @return the color sensor value as a DriverStation.Alliance enum.
+     */
+    public DriverStation.Alliance getColor() {
+        Color color = colorSensorIntake.getColor();
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+        SmartDashboard.putBoolean("match", resultColor == Color.kRed);
+
+        if (resultColor == Constants.Conveyor.RED) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Constants.Conveyor.BLUE) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    /**
+     * @return the amount of balls inside the cargo.
+     */
+    public int getCargoCount() {
+        return position.size();
+    }
+
+    /**
+     * Set motor power.
+     *
+     * @param power the power to the motor [-1, 1].
+     */
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    /**
+     * open the "flap" - solenoid
+     */
+    public void openFlap() {
+        flap.set(false);
+    }
+
+    /**
+     * closes the "flap" - solenoid
+     */
+    public void closeFlap() {
+        flap.set(true);
+    }
+
+    /**
+     * toggles the "flap" - solenoid
+     */
+    public void toggle() {
+        flap.toggle();
+    }
+
+    private void updateActualBallPositions() {
+        var colorIntake = getColor();
+        boolean currentPostFlapBeamInput = postFlapBeam.get();
+        SmartDashboard.putString("alliance", colorIntake.name());

```

Comment:
"alliance" sounds like the color of the alliance _Stardust_ plays in. That's not the case here though.